bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -154,17 +154,33 @@ type RKE2AgentConfig struct {
 	// AdditionalUserData is a field that allows users to specify additional cloud-init or ignition configuration to be included in the
 	// generated cloud-init/ignition script.
 	//+optional
-	AdditionalUserData AdditionalUserData `json:"additionalUserData,omitempty"`
+	Ignition IgnitionSpec `json:"ignition,omitempty"`
 }
 
-// AdditionalUserData is a field that allows users to specify additional cloud-init configuration .
-type AdditionalUserData struct {
-	// In case of using ignition, the data format is documented here: https://kinvolk.io/docs/flatcar-container-linux/latest/provisioning/cl-config/
-	// NOTE: All fields of the UserData that are managed by the RKE2Config controller will be ignored, this include "write_files", "runcmd", "ntp".
+// IgnitionSpec contains Ignition specific configuration.
+type IgnitionSpec struct {
+	// ButaneConfig contains butane specific configuration.
 	// +optional
-	Config string `json:"config,omitempty"`
+	ButaneConfig *ButaneConfig `json:"butaneConfig,omitempty"`
 
-	// Strict controls if Config should be strictly parsed. If so, warnings are treated as errors.
+	// Ignition version, Default to clc
+	// +kubebuilder:validation:Enum=butane
+	// +optional
+	Transpiler string `json:"transpiler,omitempty"`
```

Comment:
Perhaps I'm misunderstanding, but today doesn't `additionalUserData` accept ignition JSON directly?  AFAICS the CLC usage is just internal templating, so we could just remove it and avoid adding this new dependency on Butane? 

@alexander-demicev perhaps you can confirm?

Reply:
`AdditionalUserData` accepts CLC config that is later transformed into Ignition. I think we can avoid changing anything in the API objects. If we want to drop support for Ignition v2, let's leave the API as it is, remove the CLC package, and add Butane to replace it on the backend.
This way there will be no breaking changes on the API object itself, only the format of `AdditionalUserData.Config` will change(butane instead of clc) and on the backend butane will be used for templating.

Reply:
we need ignition v3 for sure because Sle Micro only support v3

Reply:
sure, so we can drop v2 entirely on the backend by removing CLC and using Butane, and on the API side no changes are needed

Reply:
done!